python-checks/src/main/java/org/sonar/python/checks/BuiltinShadowingAssignmentCheck.java

```diff
@@ -86,15 +93,35 @@ private void checkAnnotatedAssignment(SubscriptionContext ctx) {
   }
 
   private void raiseIssueForNonGlobalVariable(SubscriptionContext ctx, Name variable) {
-    Symbol variableSymbol = variable.symbol();
-    if (variableSymbol != null && variableSymbol.usages().stream().noneMatch(u -> u.kind().equals(Usage.Kind.GLOBAL_DECLARATION))) {
-      PreciseIssue existingIssue = variableIssuesRaised.get(variableSymbol);
-      if (existingIssue != null) {
-        existingIssue.secondary(variable, REPEATED_VAR_MESSAGE);
-      } else {
-        variableIssuesRaised.put(variableSymbol, ctx.addIssue(variable, MESSAGE));
-      }
-    }
+    Optional.ofNullable(variable.symbol())
+      .filter(symbol -> symbol.usages().stream().map(Usage::kind).noneMatch(Usage.Kind.GLOBAL_DECLARATION::equals))
+      .ifPresent(symbol -> {
```

Comment:
I'm not sure I get why using `Optional` and a functional style is better here.
It seems to me that it has at least one drawback: it may hide test coverage issues, e.g. we may not see that we never test a case where `variable.symbol()` is null.

Reply:
That's true but in case we don't need to react somehow on a null case there - it keeps code more clear for me instead of using two `if` statements (null check and usages filtering)